homeassistant/components/broadlink/remote.py

```diff
@@ -0,0 +1,376 @@
+        return
+    hass.data[DOMAIN][COMPONENT].append(unique_id)
+
+    api = broadlink.rm((host, DEFAULT_PORT), mac_addr, None)
+    api.timeout = timeout
+    code_storage = Store(hass, CODE_STORAGE_VERSION, CODE_STORAGE_KEY.format(unique_id))
+    flag_storage = Store(hass, FLAG_STORAGE_VERSION, FLAG_STORAGE_KEY.format(unique_id))
+    remote = BroadlinkRemote(name, unique_id, api, code_storage, flag_storage)
+
+    connected, loaded = (False, False)
+    try:
+        connected, loaded = await asyncio.gather(
+            hass.async_add_executor_job(api.auth), remote.async_load_storage_files()
+        )
+    except socket.error:
+        pass
+    if not connected:
+        hass.data[DOMAIN][COMPONENT].remove(unique_id)
+        raise PlatformNotReady
+    if not loaded:
+        _LOGGER.error("Failed to set up %s", unique_id)
+        hass.data[DOMAIN][COMPONENT].remove(unique_id)
+        return
+    async_add_entities([remote], False)
+
+
+class BroadlinkRemote(RemoteDevice):
+    """Representation of a Broadlink remote."""
+
+    def __init__(self, name, unique_id, api, code_storage, flag_storage):
+        """Initialize the remote."""
+        self._name = name
+        self._unique_id = unique_id
+        self._api = api
+        self._code_storage = code_storage
+        self._flag_storage = flag_storage
+        self._codes = {}
+        self._flags = defaultdict(int)
+        self._state = True
+        self._available = True
+
+    @property
+    def name(self):
+        """Return the name of the remote."""
+        return self._name
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the remote."""
+        return self._unique_id
+
+    @property
+    def is_on(self):
+        """Return True if the remote is on."""
+        return self._state
+
+    @property
+    def available(self):
+        """Return True if the remote is available."""
+        return self._available
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_LEARN_COMMAND
+
+    @callback
+    def flags(self):
+        """Return dictionary of toggle flags.
+
+        A toggle flag indicates whether `self._async_send_code()`
+        should send an alternative code for a key device.
+        """
+        return self._flags
+
+    async def async_turn_on(self, **kwargs):
+        """Turn the remote on."""
+        self._state = True
+
+    async def async_turn_off(self, **kwargs):
+        """Turn the remote off."""
+        self._state = False
+
+    async def async_update(self):
+        """Update the availability of the remote."""
+        if not self.available:
+            await self._async_connect()
+
+    async def async_load_storage_files(self):
+        """Load codes and toggle flags from storage files."""
+        try:
+            self._codes.update(
+                await self._async_get_data(self._code_storage, CODE_STORAGE_SCHEMA)
```

Comment:
There is no need to use a schema. With storage, you're responsible for loading and storing data, so the data is always in the correct format.

Reply:
Done.